tui/src/confirmation.rs

```diff
@@ -0,0 +1,127 @@
+use std::borrow::Cow;
+
+use crate::{float::FloatContent, hint::Shortcut};
+
+use crossterm::event::{KeyCode, KeyEvent};
+use ratatui::{
+    layout::Alignment,
+    prelude::*,
+    widgets::{Block, Borders, Clear, List},
+};
+
+pub enum ConfirmStatus {
+    Confirm,
+    Abort,
+    None,
+}
+
+pub struct ConfirmPrompt {
+    pub names: Box<[String]>,
+    pub status: ConfirmStatus,
+    scroll: usize,
+}
+
+impl ConfirmPrompt {
+    pub fn new(names: &[&str]) -> Self {
+        let max_count_str = format!("{}", names.len());
+        let names = names
+            .iter()
+            .zip(1..)
+            .map(|(name, n)| {
+                let count_str = format!("{n}");
+                let space_str = (0..(max_count_str.len() - count_str.len()))
+                    .into_iter()
+                    .map(|_| ' ')
+                    .collect::<String>();
```

Comment:
Clippy is mad again.

```suggestion
                let space_str = (0..(max_count_str.len() - count_str.len()))
                    .map(|_| ' ')
                    .collect::<String>();
```

Reply:
yeah, I should just add `cargo clippy` to a precommit hook at this point

Reply:
Add `cargo clippy -- -Dwarnings` so it'll return `0` on warnings.